lib/cooked_post_processor.rb

```diff
@@ -85,6 +86,29 @@ def post_process_quotes
     end
   end
 
+  def removed_direct_reply_full_quotes
+    return if @post.post_number == 1
+
+    num_quotes = @post.raw.scan(/\[quote[^\]]*\]/).size
+    return if num_quotes != 1
+
+    prev = @post.topic.posts.order('post_number desc').where('post_number < ? AND not hidden AND posts.deleted_at IS NULL', @post.post_number).limit(1).pluck(:raw).first
```

Comment:
A few notes:
- `@post.topic.posts` will generate an unnecessary query.
- `posts.deleted_at IS NULL` is already checked by the default scope of `Post`.
- Also, we only want to check for `regular` posts.

Here's what I suggest instead:

```suggestion
    prev = Post.where("topic_id = ? AND post_number < ? AND post_type = ? AND NOT hidden", @post.topic_id, @post.post_number, Post.types[:regular]).limit(1).pluck(:raw).first
```